File: codeforces/268A - Games/a.py
from collections import defaultdict
def main():
    n = int(input())
    home = []
    away = defaultdict(int)
    for _ in range(n):
        h, a = map(int, input().split())
        home.append(h)
        away[a]+=1
    clash = 0
    for i in home:
        clash += away[i]
    print(clash)
if __name__ == "__main__":
    main()

File: codeforces/268A - Games/a.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int n;
    cin >> n;
    vector<int> home;
    unordered_map<int,int> away;
    for(int i = 0; i < n; i++) {
        int h, a;
        cin >> h >> a;
        home.push_back(h);
        away[a]++;
    }
    int clash = 0;
    for(auto x: home) {
        clash += away[x];
    }
    cout << clash;
    return 0;
}